### PR TITLE
[DVDDemuxFFMPEG] Increase probesize for mpegts streams.

### LINKED ISSUE #26433: probesize needs to be larger for some .m2ts
I ran into some .m2ts files that fail to detect the audio properties unless I increase probesize.
Symptoms are 0 channels detected for DTS streams and subsequently no audio.

The diff below helped my Linux build but I have no idea whether this is the right spot / approach.
Dropping to 5 << 20 for instance fails to detect the channels again, so it does something.

```
$ git diff
diff --git a/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp b/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
index 67cbae5752..a2bbd1d3bf 100644
--- a/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
+++ b/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
@@ -484,6 +484,10 @@ bool CDVDDemuxFFmpeg::Open(const std::shared_ptr<CDVDInputStream>& pInput, bool
     if (m_pInput->IsStreamType(DVDSTREAM_TYPE_DVD))
       av_opt_set_int(m_pFormatContext, "analyzeduration", 500000, 0);
 
+    /* some m2ts take a lot of bytes to start all streams */
+    if (isBluray)
+      av_opt_set_int(m_pFormatContext, "probesize", 10 << 20, 0);
+
     CLog::Log(LOGDEBUG, "{} - avformat_find_stream_info starting", __FUNCTION__);
     int iErr = avformat_find_stream_info(m_pFormatContext, NULL);
     if (iErr < 0)
```

### PR DESCRIPTION
**Pull request by 78andyp**
## Description

Increases FFMPEG probesize for M2TS streams from 5MB (internal default) to 10MB.
With some UHD discs that start an M2TS stream with an animation before the main movie, all streams are not detected in the first 5MB.

## Motivation and context

Fix #26433.

## How has this been tested?

In bluray-external slack channel (thanks to @mschilder123)
Also included in OSMC (thanks to @samnazarko)

## What is the effect on users?

Improve bluray stream detection.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by fritsch** [comment]
@ksooo please check if this kills PVR switching times.

**Comment by ksooo** [comment]
> please check if this kills PVR switching times.

Did a very brief test. Found nothing unusual.

**Review by fritsch — APPROVED** [review]
It's always hard to see if that hits another edge case, e.g. some memory issue on low memory device. A stream change later one might also not be what is intended - and knowing what comes ahead of time is also not possible. If that works ... okay.

**Comment by samnazarko** [comment]
CC @tanio99. We have had this in OSMC for months with no issues (PVR and ISO tested well).

**Comment by 78andyp** [comment]
Thanks all.
